Flutter/calendar/appointments.md

```diff
@@ -130,6 +130,129 @@ events.notifyListeners(CalendarDataSourceAction.reset, null);
+                  startTime: _controller.displayDate!,
+                  endTime:
+                      _controller.displayDate!.add(const Duration(hours: 2)),
+                  subject: 'Add Appointment',
+                  color: Colors.pink);
+              _events?.appointments!.add(app);
+              _events?.notifyListeners(
+                  CalendarDataSourceAction.add, <Appointment>[app]);
+            },
+          ),
+          body: SfCalendar(
+            view: CalendarView.month,
+            controller: _controller,
+            dataSource: _events,
+          )),
+    );
+  }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Remove Action
+
+When we remove appointments from a data source dynamically, we must notify the data source for the remove action.
+
+{% tabs %}
+{% highlight dart hl_lines="2 3 4" %}
+
+final Appointment removeAppointment = appointmentCollection[0];
+appointmentCollection.remove(removeAppointment);
+_events?.notifyListeners(CalendarDataSourceAction.remove,
+    <Appointment>[removeAppointment]);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Reset Action
+
+When we replace appointments in a data source dynamically, we must notify the data source for the reset action.
+
+{% tabs %}
+{% highlight dart hl_lines="1 2 3" %}
+
+_events?.appointments!.clear();
+_events?.notifyListeners(
+    CalendarDataSourceAction.reset, _events!.appointments!);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Add Resource
+
+When dynamically adding resource to a data source, we must notify the data source for add resource.
+
+{% tabs %}
+{% highlight dart hl_lines="6 7" %}
+
+final CalendarResource resource = CalendarResource(
+displayName: 'Sophia',
+color: Colors.red,
+id: '00045',
+);
+_events!.resources!.add(resource);
+_events!.notifyListeners(CalendarDataSourceAction.addResource,
+    <CalendarResource>[resource]);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Remove Resource
+
+When we remove resource from a data source dynamically, we must notify the data source for remove resource.
+
+{% tabs %}
+{% highlight dart hl_lines="2 4" %}
+
+final CalendarResource resource = _employeeCollection[0];
+_events!.resources!.remove(resource);
+_events!.notifyListeners(
+    CalendarDataSourceAction.removeResource,
+    <CalendarResource>[resource]);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Reset Resource
+
+When we replace resource in a data source dynamically, we must notify the data source for reset resource.
+
+{% tabs %}
+{% highlight dart hl_lines="5 8" %}
+
+final List<CalendarResource> collection =
+    <CalendarResource>[];
+collection.add(CalendarResource(
+    displayName: "Sophia", id: '00045', color: Colors.green));
+_events!.resources!.clear();
+_events!.resources!.addAll(collection);
+_events!.notifyListeners(
+    CalendarDataSourceAction.resetResource, collection);
+
+{% endhighlight %}
+{% endtabs %}
+
 ## Creating business objects
 
 You can create a custom class `Meeting` with mandatory fields `from`, and `to`.
```

Comment:
You can create a custom class `Meeting` with mandatory fields `from` and `to.`

Reply:
I have checked and changed the content. I have updated my changes in the latest commit.